Sources/VLCLibraryViewController.m

```diff
@@ -39,6 +40,20 @@
 
 @implementation EmptyLibraryView
 
+- (id)initWithFrame:(CGRect)frame isInFolder:(bool)inFolder
+{
+    if (self) {
+        NSArray *subviewArray = [[NSBundle mainBundle] loadNibNamed:@"VLCEmptyLibraryView" owner:self options:nil];
+        self = [subviewArray objectAtIndex:0];
```

Comment:
you should always use .firstObject because it will give you nil if there are not subviews.
Right now this would crash if something goes wrong with loading the Nib

Reply:
This doesn't look correct to me. Shouldn't it be if (!self) and then you initialize? I need to take another look but it feels to me like this code doesn't belong here as well

Reply:
My 5 cents:
Usually override of init require call 
self = [super init];
In this case self will be setup properly and we need to check for "if (self)".
In case if we need load view from xib file the common solution is to make static method to do this. For example:
```
+ (instanceType)emptyLibraryViewInFolder:(bool)inFolder
{
    NSArray *subviewArray = [[NSBundle mainBundle] loadNibNamed:@"VLCEmptyLibraryView" owner:self options:nil];
    EmptyLibraryView *result = subviewArray.firstObject;
    [result setupEmptyValuesInFolder:inFolder];
    return result;
}

- (void)setupEmptyValuesInFolder:(bool)inFolder
{
    _emptyLibraryLabel.text = inFolder ? NSLocalizedString(@"FOLDER_EMPTY", nil) : NSLocalizedString(@"EMPTY_LIBRARY", nil);
    _emptyLibraryLongDescriptionLabel.text = inFolder ? NSLocalizedString(@"FOLDER_EMPTY_LONG", nil) : NSLocalizedString(@"EMPTY_LIBRARY_LONG", nil);
    [_learnMoreButton setTitle:NSLocalizedString(@"BUTTON_LEARN_MORE", nil) forState:UIControlStateNormal];
    _learnMoreButton.hidden = inFolder;
}

```

Reply:
thumbs up for the mentioning of super